File: Python_Solutions/palindrome/palindrome.py
# Given a string, return true
# if the string is a palindrome

def palindrome(string):
    i = 0
    while(i <= len(string)-(i+1)):
        if(string[i] != string[len(string) - (i+1)]):
            return False

        i = i + 1
    return True


print(palindrome('baab'))

# Alternate solution 1
# def palindrome(string):
#     rev = ''

#     for i in range(len(string)-1, -1, -1):
#         rev = rev + string[i]

#     return rev == string

File: Js_Solutions/palindrome/palindrome.js
// Given a string, return true
// if the string is a palindrome


function palindrome(str) {
    let i = 0;
    while (i <= str.length - (i + 1)) {
        if (str[i] != str[str.length - (i + 1)]) {
            return false;
        }

        i++;
    }
    return true;
}


console.log(palindrome('baaab'))

// Alternate solution
// function palindrome(str) {
//     rev = '';

//     for (let i = str.length - 1; i >= 0; i--) {
//         rev = rev + str[i];
//     }

//     return str === rev;
// }